### PR TITLE
docs: update plugin subcommands documentation for v3

### PR DESCRIPTION
**Pull request by UndiedGamer**
> **Warning**
> *This pr should **only** be merged once [Subcommands rewrite](https://github.com/sapphiredev/plugins/pull/271/) is published to npm*

### CODE REVIEW

File: docs/Guide/plugins/Subcommands/getting-started.mdx [hunk 1]
```diff
@@ -16,29 +16,54 @@ npm install @sapphire/plugin-subcommands @sapphire/framework @sapphire/utilities
...
+  ChatInputSubcommandMappings,
+  SubcommandMappingsArray
+} from '@sapphire/plugin-subcommands';
 import type { Args } from '@sapphire/framework';
-import type { Message } from 'discord.js';
+import type { Message, CommandInteraction } from 'discord.js';
 
 // Extend `SubCommandPluginCommand` instead of `Command`
 export class UserCommand extends SubCommandPluginCommand {
-  public constructor(context: SubCommandPluginCommand.Context, options: SubCommandPluginCommand.Options) {
```
Comment (c43721) [inline docs/Guide/plugins/Subcommands/getting-started.mdx:26]: curious on the removal of `public` here as it's used throughout the rest of the file

File: docs/Guide/plugins/Subcommands/subcommand-groups.mdx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  SubCommandPluginCommand,
+  MessageSubcommandGroupMappings,
+  SubcommandMappingsArray
+} from '@sapphire/plugin-subcommands';
+import type { Args } from '@sapphire/framework';
+import type { Message } from 'discord.js';
+
+// Extend `SubCommandPluginCommand` instead of `Command`
+export class UserCommand extends SubCommandPluginCommand {
+  constructor(context, options) {
```
Comment (c43721) [inline docs/Guide/plugins/Subcommands/subcommand-groups.mdx:23]: another case of `public` versus bare constructor

**Review by c43721 — CHANGES_REQUESTED** [review]
Small feedback regarding `[public] constructor`

File: docs/Guide/plugins/Subcommands/getting-started.mdx (hunk 1 above)
Comment (UndiedGamer) [reply docs/Guide/plugins/Subcommands/getting-started.mdx:26]: added it back

**Review by c43721 — APPROVED** [review]

**Comment by favna** [comment]
Converted to draft for the time being

**Comment by UndiedGamer** [comment]
@favna could you please add the "blocked" label

File: docs/Guide/plugins/Subcommands/getting-started.mdx [hunk 3]
```diff
@@ -16,29 +16,54 @@ npm install @sapphire/plugin-subcommands @sapphire/framework @sapphire/utilities
...
 import type { Args } from '@sapphire/framework';
-import type { Message } from 'discord.js';
+import type { Message, CommandInteraction } from 'discord.js';
 
-// Extend `SubCommandPluginCommand` instead of `Command`
-export class UserCommand extends SubCommandPluginCommand {
-  public constructor(context: SubCommandPluginCommand.Context, options: SubCommandPluginCommand.Options) {
+// Extend `Subcommand` instead of `Command`
+export class UserCommand extends Subcommand {
+  public constructor(context, options) {
```
Comment (favna) [inline docs/Guide/plugins/Subcommands/getting-started.mdx:26]: types

File: docs/Guide/plugins/Subcommands/getting-started.mdx [hunk 4]
```diff
@@ -16,29 +16,54 @@ npm install @sapphire/plugin-subcommands @sapphire/framework @sapphire/utilities
...
-  public async show(message: Message, args: Args) {}
+  public async chatInputRemove(interaction: CommandInteraction) {}
+
+  public async chatInputList(interaction: CommandInteraction) {}
 }
 ```
+
+:::note
+
+To use subcommands on slash commands, you need to
```
Comment (favna) [inline docs/Guide/plugins/Subcommands/getting-started.mdx:65]:
```suggestion
To use subcommands on Chat Input Commands, you need to
```

File: docs/Guide/plugins/Subcommands/getting-started.mdx [hunk 5]
```diff
@@ -16,29 +16,54 @@ npm install @sapphire/plugin-subcommands @sapphire/framework @sapphire/utilities
 
 ### Usage
 
-```typescript ts2esm2cjs|{7-8}|{7-8}
-import { SubCommandPluginCommand } from '@sapphire/plugin-subcommands';
+```typescript ts2esm2cjs
```
Comment (favna) [inline docs/Guide/plugins/Subcommands/getting-started.mdx:19]: Re-add highlight on the subcommands array

File: docs/Guide/plugins/Subcommands/subcommand-groups.mdx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+sidebar_position: 2
+---
+
+Before we start, we assume you already know what subcommand groups are, if not please refer to the
+[Official Discord Docs](https://discord.com/developers/docs/interactions/application-commands#subcommands-and-subcommand-groups)
+for more information.
+
+## Message Subcommand Groups
+
+```typescript ts2esm2cjs
```
Comment (favna) [inline docs/Guide/plugins/Subcommands/subcommand-groups.mdx:12]: Add highlights

File: docs/Guide/plugins/Subcommands/subcommand-groups.mdx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+```
+
+## Chat Input Subcommand Groups
+
+```typescript ts2esm2cjs
+import {
+  Subcommand,
+} from '@sapphire/plugin-subcommands';
+import type { Args } from '@sapphire/framework';
+import type { CommandInteraction } from 'discord.js';
```
Comment (favna) [inline docs/Guide/plugins/Subcommands/subcommand-groups.mdx:51]: Use `Subcommand.ChatInputInteraction`

**Review by favna — CHANGES_REQUESTED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**